File: tauri-app/src-tauri/src/models/platform.rs
use serde::Serialize;

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CodexHomeStatus {
    pub path: String,
    pub exists: bool,
    pub source: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct PlatformCapabilities {
    pub platform: String,
    pub shell: String,
    pub floating_window: PlatformFeatureCapability,
    pub floating_transparency: PlatformFeatureCapability,
    pub floating_drag: PlatformFeatureCapability,
    pub floating_lock: PlatformFeatureCapability,
    pub status_tray: PlatformFeatureCapability,
    pub status_tray_live_text: PlatformFeatureCapability,
    pub autostart: PlatformFeatureCapability,
    pub notifications: PlatformFeatureCapability,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct PlatformFeatureCapability {
    pub available: bool,
    pub status: String,
    pub label: String,
    pub note: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct AutostartStatus {
    pub available: bool,
    pub enabled: bool,
    pub status: String,
    pub message: String,
}

File: tauri-app/src/types/platform.ts
export interface CodexHomeStatus {
  path: string;
  exists: boolean;
  source: string;
}

export interface CodexHomeSourceEnvelope {
  codexHome: CodexHomeStatus;
  canonicalHomeKey: string;
  physicalHomeKey: string;
  transitionGeneration: number;
}

export interface CodexHomeSourceToken {
  canonicalHomeKey: string;
  physicalHomeKey: string;
  transitionGeneration: number;
}

export interface PlatformCapabilities {
  platform: string;
  shell: string;
  floatingWindow: PlatformFeatureCapability;
  floatingTransparency: PlatformFeatureCapability;
  floatingDrag: PlatformFeatureCapability;
  floatingLock: PlatformFeatureCapability;
  statusTray: PlatformFeatureCapability;
  statusTrayLiveText: PlatformFeatureCapability;
  autostart: PlatformFeatureCapability;
  notifications: PlatformFeatureCapability;
}

export interface PlatformFeatureCapability {
  available: boolean;
  status: "ready" | "pending" | "unavailable" | string;
  label: string;
  note: string;
}

export interface AutostartStatus {
  available: boolean;
  enabled: boolean;
  status: "enabled" | "disabled" | "unavailable" | string;
  message: string;
}
